### PR TITLE
chore: Add additional data to already claimed links and already paid requests

### PR DESCRIPTION
**Pull request by panosfilianos**
Previously, the claimed link view and already paid request link view did not include all data available. This resulted in poorer UX. This PR:
- Adds extra data in Claimed Link View
- Adds extra data in Already Paid Request Link View
- Changes tab titles to be more specific in both views

### CODE REVIEW

**Comment by Hugo0** [comment]
2 small nits:

- 1. PR would benefit from motivation for PR in the description (short 1-2sentences)
- 2. I'd say this is more of a chore than a fix, but honestly it doesn't rlly matter

File: src/app/api/peanut/get-link-info/route.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import type { NextRequest } from 'next/server'
+import { NextResponse } from 'next/server'
+import * as consts from '@/constants'
+
+export async function POST(request: NextRequest) {
+    try {
+        const { link } = await request.json()
+
+        const response = await fetch(`${consts.PEANUT_API_URL}/get-link-info`, {
```
Comment (jjramirezn) [inline src/app/api/peanut/get-link-info/route.ts]: comment(blocking): the existing endpoint is /get-link-details

Also where is used this new route? I don't see it used in this PR

File: src/app/api/peanut/get-link-info/route.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import type { NextRequest } from 'next/server'
+import { NextResponse } from 'next/server'
+import * as consts from '@/constants'
+
+export async function POST(request: NextRequest) {
+    try {
+        const { link } = await request.json()
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply src/app/api/peanut/get-link-info/route.ts]: valid

File: src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx [hunk 3]
```diff
@@ -7,21 +7,72 @@ import * as utils from '@/utils'
...
-                        ?.tokens.find((token) =>
-                            utils.areTokenAddressesEqual(token.address, requestLinkData?.tokenAddress ?? '')
-                        )?.symbol ?? requestLinkData?.tokenAddress}{' '}
-                on{' '}
-                {consts.supportedPeanutChains &&
-                    consts.supportedPeanutChains.find((chain) => chain.chainId == requestLinkData?.chainId)?.name}
-            </label>
+            {dataAvailable && (
+                <div className="flex w-full flex-col items-center justify-center gap-2">
+                <div className="flex w-full flex-col items-center justify-center gap-2">
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx]: look into this

File: src/app/api/peanut/get-link-info/route.ts (hunk 1 above)
Comment (panosfilianos) [reply src/app/api/peanut/get-link-info/route.ts]: to be removed based on comments from @Hugo0

File: src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx [hunk 4]
```diff
@@ -7,21 +7,72 @@ import * as utils from '@/utils'
...
+    const chainAvailable: boolean = requestLinkData?.chainId != undefined
+    const recipientAddressAvailable: boolean = requestLinkData?.recipientAddress != undefined
+    const dataAvailable: boolean = tokenSymbolAvailable || tokenAmountAvailable || chainAvailable || recipientAddressAvailable
+    
     return (
         <div className="flex w-full flex-col items-center justify-center gap-6 py-2 pb-20 text-center">
             <label className="text-h2">Sorry, this link has already been paid.</label>
-            <label className="text-h8 font-bold ">
-                This link previously requested{' '}
-                {requestLinkData?.tokenSymbol ??
```
Comment (jjramirezn) [inline src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx:15]: thought: there is loss of functionality here

Before we were trying to get a token symbol if there is none from the data. Now we just trust the data we get. I actually like the new way better but it warrants fixing why sometimes we don't have data from the request link itself. In the short term this change WILL result in some data previously shown to not be shown anymore

File: src/utils/link.utils.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+// This file holds all relevant link utils
+// TODO: bring all link utils from general.utils.ts to here
+
+export async function get(bankAccount: string): Promise<boolean> {
```
Comment (jjramirezn) [inline src/utils/link.utils.ts]: comment(blocking): where is this used?

also, we should probably use better naming

File: src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx (hunk 4 above)
Comment (panosfilianos) [reply src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx:15]: True. I would expect this operation to take place where data is populated (i.e. where requestLinkData is formed, populate as lines 16-19 do). I, actually, do not have a clear idea on why requestLinkData won't have all required fields populated already. Thus, I opted to show the data that is directly available through requestLinkData.

**Review by jjramirezn — CHANGES_REQUESTED** [review]
- The function in link.utils.ts seems to be unused
- Fix deployment (using pnpm build you can check whats failing, but there is a duplicated div tag maybe is that)

**Comment by panosfilianos** [comment]
New updates:
- Removing unused functions (previously added to fetch additional link data, which halfway we decided was not needed)
- `pnpm run dev` executes properly
- Removed uneeded div
Thanks a lot for the comments @jjramirezn

**Comment by jjramirezn** [comment]
pnpm dev compiles lazily and I don't know if checks types, the error that shows with `pnpm build` is the following:



Also I think is better not to commit the empty links.utils.ts file (when we start moving some functions there we can create it)

**Review by jjramirezn — APPROVED** [review]
Approved, merging. 

Nice PR, Looks much better now!

**Comment by panosfilianos** [comment]
- Fixed minor build bug
- Removed utils file (to add a Task from TODO: moving utils functions around links on their own file)
- Vercel deploys successfully

**Pull request merged**